src/blueprints/Do.Blueprints.Service.Application/Domain/Model/TypeModel.cs

```diff
@@ -1,21 +1,28 @@
-﻿using System.Reflection;
-
-namespace Do.Domain.Model;
+﻿namespace Do.Domain.Model;
 
 public record TypeModel(
-    Type Type,
     string Name,
-    List<MethodModel> Methods
-)
+    string? Namespace,
+    ModelCollection<ConstructorModel> Constructors,
+    ModelCollection<MethodModel> Methods,
+    ModelCollection<PropertyModel> Properties,
+    ModelCollection<TypeModel> GenericTypeArguments,
+    ModelCollection<TypeModel> CustomAttributes,
+    bool IsAbstract,
+    bool IsValueType
+) : IModel
 {
+    readonly Type _type = default!;
+    readonly string _id = default!;
+
     public TypeModel(Type type)
-        : this(type, type.Name, [])
+        : this(type.Name, type.Namespace, [], [], [], [], [], type.IsAbstract, type.IsValueType)
     {
-        var methodInfos = type.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly) ?? [];
-
-        foreach (var method in methodInfos)
-        {
-            Methods.Add(new(method));
-        }
+        _type = type;
+        _id = IModel.IdFromType(type);
     }
+
+    public string Id => _id;
+
+    internal void Apply(Action<Type> action) => action(_type);
```

Comment:
make this public

Reply:
you are ignoring the fact that we will distribute these classes in separate packages. domain model can't depend on fluent nhibernate, because domain layer will probably be in do.architecture, and orm and many other features will probably need to have access to this apply method to make configurations